sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -169,6 +174,14 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
   private def tryParseDouble(field: String): DataType = {
     if ((allCatch opt field.toDouble).isDefined || isInfOrNan(field)) {
       DoubleType
+    } else {
+      tryParseDateTime(field)
+    }
+  }
+
+  private def tryParseDateTime(field: String): DataType = {
+    if ((allCatch opt dateFormatter.parse(field)).isDefined) {
```

Comment:
We should probably 1. add  either SQL configuration or an option e.g., `infersDate` (like `prefersDecimal` in JSON options), or 2. only infers date types when `dateFormat` is specified.  The main idea is to avoid perf regression in schema inference by adding this.